src/App.tsx

```diff
@@ -1,26 +1,122 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState, useRef } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import * as todoService from './api/todos';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+import { Error } from './types/Error';
+import { TodoList } from './components/TodoList';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoItem } from './components/TodoItem';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isInputLoading, setIsInputLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState(Error.none);
+  const [filter, setFilter] = useState(Filter.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [deletingTodoIds, setDeletingTodoIds] = useState<number[]>([]);
+
+  const textField = useRef<HTMLInputElement>(null);
+
+  const completedTodos = [...todos].filter(todo => todo.completed);
+  const activeTodos = [...todos].filter(todo => !todo.completed);
```

Comment:
```suggestion
  const activeTodos = todos.filter(todo => !todo.completed);
```
the filter method doesn't mutate the array